### DISCUSSION TITLE
Watched value and handleSubmit value out of sync when using setValue

### DISCUSSION THREAD

**Discussion by Fumler** [react-hook-form/react-hook-form#5821] [Q&A] [answered]
**Describe the bug**
When using `setValue` in a controlled component the value seems to be updated in values seen by `watch`, but when then calling `handleSubmit` the data is the previous value, as if the form has been reset, it's kind of hard to explain so reproducing it will probably make it easier to understand.

Not sure if this is just me doing something wrong or if there's a bug somewhere here, but would appreciate any help. Sorry for the very contrived example as well, the original source is in a private repo.

**To Reproduce**

1. View the current form data, e.g. you will see the `openlibrary` key has the value `["OL22853304M"]`
2. Click on `Remove` on the `openlibrary` item
3. View the current form data in devtools or console, you will see the value has been updated to `[]`
4. Click the bottom `submit` button.
5. View the `console.log` from the submit and you will see `openlibrary` has it's original value, while devtools still has the new value

**Codesandbox link (Required)**
https://codesandbox.io/s/focused-satoshi-tekj5?file=/src/App.tsx

**Expected behavior**
I expect the data I set in `setValue` to be updated when calling `handleSubmit`

**Desktop (please complete the following information):**

- OS: Windows 10 (WSL2, Ubuntu 20.04)
- Browser: Sidekick (Chromium)
- Version: 90.9.1.9579

**Additional context**
Using v7.10.0 of react-hook-form

**Comment by bluebill1049** [comment, marked as answer]
hi there, not sure if I am following the issue here.

I have removed the dev tool and compared the watch vs handleSubmit (looks fine)

<img width="655" alt="Screen Shot 2021-07-03 at 10 50 24 am" src="https://user-images.githubusercontent.com/10513364/124338648-79f81b80-dbec-11eb-9cab-91f4a547dcb7.png">
<img width="593" alt="Screen Shot 2021-07-03 at 10 50 55 am" src="https://user-images.githubusercontent.com/10513364/124338664-8d0aeb80-dbec-11eb-972e-f45068d3d480.png">

**Reply by Fumler** [reply]
Created a video so it's perfectly clear how to reproduce:

https://user-images.githubusercontent.com/467326/124353033-390c0100-dc04-11eb-9d39-3a1519e938ce.mp4

As you can see the call to `setValue` updates the watched value to be an empty array, but when you click submit the data is still the previous value.

**Reply by bluebill1049** [reply]
I reckon it's probably an issue which caused by the devTool, can you disable the dev tool and give it another try?

**Reply by Fumler** [reply]
> I reckon it's probably an issue which caused by the devTool, can you disable the dev tool and give it another try?

The video is without devTool.

**Reply by bluebill1049** [reply]
https://user-images.githubusercontent.com/10513364/124369765-cad24780-dcb2-11eb-952c-286f66b66a3e.mov

I did the same without the issue above. can you confirm my steps is correct?

**Reply by Fumler** [reply]
You are not clicking submit to trigger `handleSubmit`.

**Reply by bluebill1049** [reply]
looked closer at your example, couple things I would suggest here: 

- probably should consider using `useFieldArray`
- use Controller probably should have an input to associated with it.

The issue is mostly due to Controller try to connect with data with got `reseted`

i have changed to `register` and works ok now:

https://codesandbox.io/s/bitter-mountain-py7vw?file=/src/App.tsx

**Reply by Fumler** [reply]
Interesting. `register` does solve the issue, I always thought that's some of what `Controller` does under the hood, but good to know there is a difference here. I started with `useFieldArray` but had some issues with it, but I'll give it another go now that I know this is at least solvable with `register`.

Thank you very much for looking into it!

**Reply by bluebill1049** [reply]
no worries. let me know what are the issues with useFieldArray, happy to help.

**Comment by Seke1412** [comment]
I also encounter the same issue

With 7.10.0
![Screen Shot 2021-07-03 at 7 26 30 PM](https://user-images.githubusercontent.com/3940350/124352720-b7809700-dc34-11eb-9933-259f0f93f70d.png)

With 7.9.0 and below
![Screen Shot 2021-07-03 at 7 26 21 PM](https://user-images.githubusercontent.com/3940350/124352727-c9fad080-dc34-11eb-952f-490c49124c5d.png)

**Reply by bluebill1049** [reply]
can you provide a separate issue on this? I don't see that they r the same (with the current issue).
